clients/cobol-lsp-vscode-extension/src/__tests__/ZoweApiTest.ts

```diff
@@ -12,25 +12,131 @@
  *   Broadcom, Inc. - initial API and implementation
  */
 
-import { BasicProfileManager, Session } from "@zowe/imperative";
+import { RestClient } from "@zowe/imperative";
+import { Session } from "inspector";
+import { homedir } from "os";
+import * as path from "path";
 import { ZoweApi } from "../services/ZoweApi";
 import { Type, ZoweError } from "../services/ZoweError";
-jest.mock("@zowe/imperative");
+
+const loadMock = jest.fn();
+const loadAllMock = jest.fn();
+jest.mock("@zowe/imperative/lib/profiles/src/BasicProfileManager", () => {
+    return {
+        BasicProfileManager: jest.fn().mockImplementation(() => {
+            return {
+                load: loadMock,
+                loadAll: loadAllMock,
+                getDefaultProfileName: () => "defaultName",
+            };
+        }),
+    };
+});
 
 beforeEach(() => {
-    (BasicProfileManager as any).mockClear();
+    loadMock.mockClear();
+    loadAllMock.mockClear();
 });
 
 describe("ZoweApi", () => {
+    it("listZOSMFProfiles returns map of founded profiles", async () => {
+        const profile = {
+            name: "foo",
+            password: "bar",
+        };
+        const profileFound = {
+            name: "profileName",
+            failNotFound: true,
+            profile: profile,
+        };
+        const profileNotFound = {
+            name: "wrongName",
+            failNotFound: false,
+        };
+        loadAllMock.mockResolvedValue([profileFound, profileNotFound]);
+        const zoweApi = new ZoweApi();
```

Comment:
const zoweApi can reused to get out the duplication, via initializing it in beforeEach

Reply:
Make sense. Thanks.